### PR TITLE
Fix dropdown menu and fallback rule

### PR DESCRIPTION
**Pull request by devkanro**

### CODE REVIEW

File: core/src/main/kotlin/org/jetbrains/jewel/painter/PainterHintsProvider.kt [hunk 1]
```diff
@@ -18,6 +18,15 @@ import org.jetbrains.jewel.painter.hints.HiDpi
 @Immutable
 interface PainterHintsProvider {
 
+    /**
+     * Those hints will be added before the [hints][PainterProvider.getPainter] that passed in [PainterProvider.getPainter].
```
Comment (rock3r) [inline core/src/main/kotlin/org/jetbrains/jewel/painter/PainterHintsProvider.kt:22]:
```suggestion
     * These hints will be consumed before the [hints] when [PainterProvider.getPainter] is called.
```

File: core/src/main/kotlin/org/jetbrains/jewel/painter/PainterHintsProvider.kt [hunk 2]
```diff
@@ -18,6 +18,15 @@ import org.jetbrains.jewel.painter.hints.HiDpi
...
+     */
+    @Composable
+    fun priorityHints(path: String): List<PainterHint> = emptyList()
+
+    /**
+     * Those hints will be added after the [hints][PainterProvider.getPainter] that passed in [PainterProvider.getPainter].
```
Comment (rock3r) [inline core/src/main/kotlin/org/jetbrains/jewel/painter/PainterHintsProvider.kt:28]:
```suggestion
     * These hints will be consumed after the [priorityHints] when [PainterProvider.getPainter] is called.
```

File: core/src/main/kotlin/org/jetbrains/jewel/painter/PainterHintsProvider.kt [hunk 3]
```diff
@@ -18,6 +18,15 @@ import org.jetbrains.jewel.painter.hints.HiDpi
 @Immutable
 interface PainterHintsProvider {
 
+    /**
+     * These hints will be consumed before the [hints] when [PainterProvider.getPainter] is called.
```
Comment (devkanro) [inline core/src/main/kotlin/org/jetbrains/jewel/painter/PainterHintsProvider.kt:22]:
```suggestion
     * These hints will be consumed before the hints which are passed as a parameter to [PainterProvider.getPainter].
```

File: core/src/main/kotlin/org/jetbrains/jewel/painter/PainterHintsProvider.kt [hunk 4]
```diff
@@ -18,6 +18,15 @@ import org.jetbrains.jewel.painter.hints.HiDpi
...
+     */
+    @Composable
+    fun priorityHints(path: String): List<PainterHint> = emptyList()
+
+    /**
+     * These hints will be consumed after the [priorityHints] when [PainterProvider.getPainter] is called.
```
Comment (devkanro) [inline core/src/main/kotlin/org/jetbrains/jewel/painter/PainterHintsProvider.kt:28]:
```suggestion
     * These hints will be consumed after the hints which are passed as a parameter to [PainterProvider.getPainter].
```

**Comment by rock3r** [comment]
CI is red but that's not caused by this PR

**Review by rock3r — APPROVED** [review]

**Pull request merged**